templates/CRM/Member/Form/MembershipRenewal.tpl

```diff
@@ -270,15 +274,34 @@
       }
 
       var term = cj("#num_terms").val();
-      if (term) {
-        var renewTotal = allMemberships[memType]['total_amount_numeric'] * term;
-        cj("#total_amount").val(CRM.formatMoney(renewTotal, true));
-      }
-      else {
-        cj("#total_amount").val(allMemberships[memType]['total_amount']);
+      var taxAmount = allMemberships[memType]['tax_amount'];
+      var totalAmountExTax = allMemberships[memType]['total_amount_ex_tax'];
+      var taxRate = allMemberships[memType]['tax_rate'];
+      var currencySymbol = {/literal}{$currency_symbol|@json_encode}{literal};
+
+      if (typeof manualAmountUpdate !== "undefined" && manualAmountUpdate) {
+        if (taxRate) {
+          var totalAmountIncTax = Number(cj("#total_amount").val());
+          taxAmount = (totalAmountIncTax - ((totalAmountIncTax * 100) / Number(100 + Number(taxRate)))).toFixed(2);
+        }
+      } else {
+        if (term) {
+          var renewTotal = allMemberships[memType]['total_amount_numeric'] * term;
+          totalAmountExTax = totalAmountExTax * term;
+          cj("#total_amount").val(CRM.formatMoney(renewTotal, true));
+        }
+        else {
+          cj("#total_amount").val(allMemberships[memType]['total_amount']);
+        }
+
+        taxAmount = Number(totalAmountExTax * (taxRate / 100)).toFixed(2);
       }
 
-      cj('.totaltaxAmount').html(allMemberships[memType]['tax_message']);
+      var taxMsg = '';
+      if(!isNaN(taxAmount) && taxRate) {
+        taxMsg = 'Includes '+allMemberships[memType]['tax_term']+ ' of '+currencySymbol + ' ' + taxAmount;
+      }
+      cj('.totaltaxAmount').html(taxMsg);
```

Comment:
Can we do some consolidation here on Membership.tpl and MembershipRenewal.tpl?  Even if it's just extracting some of the common functionality into a separate tpl for now and then including it in both.

Reply:
Right we need this to be at very worse no more duplication to get it merge ready. In general I would expect it to reduce duplication - I would normally do a handful of preliminary PRs to that effect as a first step